xbmc/interfaces/builtins/SkinBuiltins.cpp

```diff
@@ -104,6 +104,49 @@ static int SetAddon(const std::vector<std::string>& params)
   return 0;
 }
 
+/*! \brief Select and choose a value for a skin setting.
+ *  \param params The parameters.
+ *  \details params[0] = Names of skin settings.
+ */
+static int SkinSelect(const std::vector<std::string>& params)
+{
+  std::vector<std::pair<std::string, std::string>> settings;
+
+  auto pDlgSelect = static_cast<CGUIDialogSelect*>(CServiceBroker::GetGUI()->GetWindowManager().GetWindow(WINDOW_DIALOG_SELECT));
+  pDlgSelect->Reset();
+  pDlgSelect->SetHeading(CVariant{atoi(params[0].c_str())});
+
+  for (auto &param : params)
+  {
+    if (param.find('|') != std::string::npos)
```

Comment:
We should skip the first two parameters before we make the loop over the parameters.  The current implementation is not very nice, because you could add a pipe to header parameter 0 and it is interpreted as a setting value.